source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,441 @@
+closeTimeout:     "number"
+Default:          "0 if using *file*, 10 if using *dynaFile*"
+Available since:  8.3.3
+================  ==============================
+
+Specifies after how many minutes of inactivity a file is
+automatically closed.
+
+Note that this functionality is implemented based on the  :doc:`janitor process <../../concepts/janitor>`.
+
+==================  ==================
+dynaFileCacheSize:  "number"
+Default:            "10"
+Legacy:             $DynaFileCacheSize
+==================  ==================
+
+**Applies only if dynamic filenames are used.**
+
+Specifies the number of files that will be kept open.
+Note that this is a per-action parameter, so if you have
+multiple dynafile actions, each of them have their individual caches
+(which means the numbers sum up).
+
+Ideally, the cache size exactly
+matches the need. You can use :doc:`impstats <impstats>` to tune
+this value. Note that a too-low cache size can be a very considerable
+performance bottleneck.
+
+=========  ===============
+zipLevel:  "number"
+Default:   "0"
+Legacy:    $OMFileZipLevel
+=========  ===============
+
+If greater than 0, turns on gzip compression of the output file. The
+higher the number, the better the compression, but also the more CPU
+is required for zipping.
+
+================  ========
+veryRobustZip:    "on|off"
+Default:          "on"
+Available since:  7.3.0
+================  ========
+
+If *zipLevel* is greater than 0,
+then this setting controls if extra headers are written to make the
+resulting file extra hardened against malfunction. 
+If set to *off*, data appended to previously unclean closed files may not be
+accessible without extra tools.
+
+**TODO Note that this risk is usually expected to be bearable, and thus "off" is the default mode.**
+
+The extra headers considerably degrade compression, files with this option set to "on" may be four to five times as large as files processed in "off" mode.
+
+==============  ====================
+flushInterval:  "seconds"
+Default:        "1"
+Legacy:         $OMFileFlushInterval
+==============  ====================
+
+Defines, in seconds, the interval after which unwritten data is flushed.
+
+=============  ===================
+asyncWriting:  "on|off"
+Default:       "off"
+Legacy:        $OMFileASyncWriting
+=============  ===================
+
+if turned *on*, the files will be written in asynchronous mode via a
+separate thread. In that case, double buffers will be used so that
+one buffer can be filled while the other buffer is being written.
+   
+Note that in order to enable *flushInterval*, this parameter must be *on*. Otherwise, the flush interval will be ignored.
+Also note that when flushOnTXEnd is *on* but asyncWriting is *off*, output will only be written when the buffer is full.
+This may take several hours, or even require a rsyslog shutdown. However, a buffer flush can be forced in that case by sending rsyslogd a HUP signal.
+
```

Comment:
asyncwriting combined with dynafile can result is a LOT of rsyslog threads (one per file currently open)